Sources/SwiftSyntaxBuilder/BuildableNodes.swift.gyb

```diff
@@ -114,6 +119,24 @@ public struct ${type.buildable()}: ${base_type.buildable()}, ${type.expressible_
   public func create${type.buildable_base_name()}() -> ${type.buildable()} {
     return self
   }
+%{
+  trailing_comma_init_args = []
+  if has_with_trailing_comma_trait:
+    for child in children:
+      if child.name() == 'trailingComma':
+        trailing_comma_init_args.append('%s: withComma ? .comma : nil' % (child.name()))
+      else:
+        trailing_comma_init_args.append('%s: %s' % (child.name(), child.name()))
+}%
+% if trailing_comma_init_args:
+
+  /// Conformance to `HasTrailingComma`.
+  public func withTrailingComma(_ withComma: Bool) -> Self {
+      return Self.init(
+        ${',\n        '.join(trailing_comma_init_args)}
+      )
+  }
+% end
```

Comment:
Is there a reason why you can’t you move the `trailing_comma_init_args` initialization into the `if` statement to make it clear it’s not being used anywhere else?

```suggestion
% if has_with_trailing_comma_trait:
%   trailing_comma_init_args = []
%   for child in children:
%     if child.name() == 'trailingComma':
%       trailing_comma_init_args.append('%s: withComma ? .comma : nil' % (child.name()))
%     else:
%       trailing_comma_init_args.append('%s: %s' % (child.name(), child.name()))

  /// Conformance to `HasTrailingComma`.
  public func withTrailingComma(_ withComma: Bool) -> Self {
      return Self.init(
        ${',\n        '.join(trailing_comma_init_args)}
      )
  }
% end
```